compiler/rustc_middle/src/ty/generics.rs

```diff
@@ -367,11 +367,12 @@ impl<'tcx> Generics {
 }
 
 /// Bounds on generics.
-#[derive(Copy, Clone, Default, Debug, TyEncodable, TyDecodable, HashStable)]
+#[derive(Copy, Clone, Debug, TyEncodable, TyDecodable, HashStable)]
 pub struct GenericPredicates<'tcx> {
     pub parent: Option<DefId>,
     pub predicates: &'tcx [(Clause<'tcx>, Span)],
     pub effects_min_tys: &'tcx ty::List<Ty<'tcx>>,
+    pub tainted_by_errors: Result<(), ErrorGuaranteed>,
```

Comment:
Explain what it means to be tainted here

Reply:
Done. If preferred I can also register a dummy predicate `{error}: Sized` or sth, that would probably result in similar behaviour